ethcore/light/src/on_demand/mod.rs

```diff
@@ -366,7 +367,7 @@ impl OnDemand {
 			.filter_map(|pending| {
 				// the peer we dispatch to is chosen randomly
 				let num_peers = peers.len();
-				let rng = rand::random::<usize>() % num_peers;
+				let rng = rand::random::<usize>() % cmp::max(num_peers, 1);
```

Comment:
Maybe it should be a question to the previous PR, but why are we always skipping the same peers from the front of `peers`?

Wouldn't it be better to randomly partition the collection instead?

Reply:
I'm not sure I get it.
The code only chooses one node to dispatch the request to (not a range of nodes). And since we go through the list of nodes twice, each node has an equal chance to be chosen if I'm not mistaken.

Reply:
I see, misread the code below. Looks good.